assets/js/modules/ads/pax/services.js

```diff
@@ -16,11 +16,35 @@
  * limitations under the License.
  */
 
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
 /**
  * Internal dependencies
  */
 import { CORE_SITE } from '../../../googlesitekit/datastore/site/constants';
 
+const restFetchWpPages = async () => {
+	try {
+		const wpPages = await apiFetch( {
+			path: '/wp/v2/pages?per_page=100',
+		} ).then( ( pages ) =>
+			pages.map( ( page ) => {
+				return {
+					title: page.title.rendered,
+					path: new URL( page.link ).pathname,
+				};
+			} )
+		);
+		if ( wpPages ) {
+			return wpPages;
+		}
```

Comment:
This condition shouldn't be necessary as a successful response should always return an array, and a non-successful response (400+) would throw. It would also leave the "else" unhandled (returning `null` implicitly), but it shouldn't be necessary anyways.

We can simplify this above without the condition to be `return wpPages.map(` to return the mapped structure rather than chaining that on the promise.

Reply:
I've simplified this as per feedback @aaemnnosttv